definitions/npm/styled-components_v4.x.x/flow_v0.92.0-/styled-components_v4.x.x.js

```diff
@@ -0,0 +1,423 @@
+    css(): string;
+    toHTML(additionalAttrs: ?string): string;
+    toElement(): React$Element<*>;
+    clone(): Tag<T>;
+    sealed: boolean;
+  }
+
+  // The `any`/weak types in here all come from `styled-components` directly, since those definitions were just copied over
+  declare export class StyleSheet {
+    static get master() : StyleSheet;
+    static get instance() : StyleSheet;
+    static reset(forceServer? : boolean) : void;
+
+    id : number;
+    forceServer : boolean;
+    target : ?HTMLElement;
+    tagMap : {[string]: Tag<any>}; // eslint-disable-line flowtype/no-weak-types
+    deferred: { [string]: string[] | void };
+    rehydratedNames: { [string]: boolean };
+    ignoreRehydratedNames: { [string]: boolean };
+    tags: Tag<any>[]; // eslint-disable-line flowtype/no-weak-types
+    importRuleTag: Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    capacity: number;
+    clones: StyleSheet[];
+
+    constructor(?HTMLElement) : this;
+    rehydrate() : this;
+    clone() : StyleSheet;
+    sealAllTags() : void;
+    makeTag(tag : ?Tag<any>) : Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    getImportRuleTag() : Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    getTagForId(id : string): Tag<any>; // eslint-disable-line flowtype/no-weak-types
+    hasId(id: string) : boolean;
+    hasNameForId(id: string, name: string) : boolean;
+    deferredInject(id : string, cssRules : string[]) : void;
+    inject(id: string, cssRules : string[], name? : string) : void;
+    remove(id : string) : void;
+    toHtml() : string;
+    toReactElements() : React$ElementType[];
+  }
+
+  declare export class KeyFrames {
+    id : string;
+    name : string;
+    rules : string[];
+
+    constructor(name : string, rules : string[]) : this;
+    inject(StyleSheet) : void;
+    toString() : string;
+    getName() : string;
+  }
+
+  // I think any is appropriate here?
+  // eslint-disable-next-line flowtype/no-weak-types
+  declare export var css : CSSConstructor;
+  declare export var keyframes : KeyFramesConstructor;
+  declare export var createGlobalStyle : CreateGlobalStyleConstructor
+  declare export var ThemeProvider: React$ComponentType<{children?: ?React$Node, theme: mixed | (mixed) => mixed}>
+
+  // This is a bit hard to read. Not sure how to make it more readable. I think adding line-breaks makes it worse.
+  declare type ThemeProps<T> = {
+    theme: T
+  }
+
+  declare export function withTheme<Theme, Config: {}, Instance>(Component: React$AbstractComponent<Config, Instance>): React$AbstractComponent<$Diff<Config, ThemeProps<Theme | void>>, Instance>
+
+  declare export type StyledComponent<Props, Theme, Instance> = React$AbstractComponent<Props, Instance> & InterpolatableComponent
+  declare interface StyledFactory<StyleProps, OwnProps, Theme, Instance> {
+    [[call]]: TaggedTemplateLiteral<StyleProps & OwnProps & ThemeProps<Theme>, StyledComponent<StyleProps & OwnProps, Theme, Instance>>;
```

Comment:
Maybe we should use `{...T1, ...T2}` instead of `T1 & T2` because of [intersection behaviour on exact object types](https://flow.org/en/docs/types/objects/#toc-exact-object-types). I tried a quick search and replace locally (maybe I effed up), but got errors; I'll double-check tomorrow morning.

Reply:
I also saw some weirdness with trying to use spread with these defs at one point. I’ll double check when I’m at my computer next to see if that’s still the case.

Reply:
I am also getting errors when attempting to use spread. I think you're right that we should be using the spread syntax, but we'll have to see what it will take to make that work...

Reply:
Should be fixed in 
45f8151